src/main/java/backend/resource/MultiModel.java

```diff
@@ -13,6 +13,9 @@
 import java.util.stream.Collectors;
 
 /**
+ * This represents the true state of local repositories data. Operations meant to change the state
+ * of local repositories data but do not go through the methods in this class i.e. operating on
+ * dangling references its sub-components are considered unsafe
```

Comment:
Okay. Someday we should enforce the latter part by copying or something. :stuck_out_tongue: